apps/mull-api/src/app/channel/channel.service.spec.ts

```diff
@@ -1,50 +1,128 @@
-import { Channel } from '../entities';
-import { mockAllChannels, mockCreateChannel } from './channel.mockdata';
+import { Repository } from 'typeorm';
+import { Channel, DirectMessageChannel, EventChannel } from '../entities';
+import { mockAllUsers } from '../user/user.mockdata';
+import { MockType } from '../user/user.service.spec';
+import {
+  mockAllChannels,
+  mockAllDirectMessageChannels,
+  mockAllEventChannels,
+  mockCreateDmChannel,
+  mockCreateEventChannel,
+} from './channel.mockdata';
 import { ChannelService } from './channel.service';
-import { CreateChannelInput } from './inputs/channel.input';
+import { CreateDmChannelInput, CreateEventChannelInput } from './inputs/channel.input';
 
 const mockChannelRepository = () => ({
   findOne: jest.fn((channelId: number) => {
     return mockAllChannels.find((channel) => channel.id === channelId);
   }),
-  save: jest.fn((mockChannelData: CreateChannelInput) => ({ ...mockChannelData })),
   delete: jest.fn((channelId: number) =>
     mockAllChannels.find((channel) => channel.id === channelId)
   ),
 });
 
+const mockEventChannelRepository = () => ({
+  findOne: jest.fn((channelId: number) => {
+    return mockAllEventChannels.find((channel) => channel.id === channelId);
+  }),
+  save: jest.fn((mockChannelData: CreateEventChannelInput) => ({ ...mockChannelData })),
+});
+
+const mockDmChannelRepository = () => ({
+  findOne: jest.fn((channelId: number) => {
+    return mockAllDirectMessageChannels.find((channel) => channel.id === channelId);
+  }),
+  save: jest.fn((mockChannelData: CreateDmChannelInput) => ({ ...mockChannelData })),
+  createQueryBuilder: jest.fn(() => ({
+    where: jest.fn().mockReturnThis(),
+    leftJoinAndSelect: jest.fn().mockReturnThis(),
+    getMany: jest.fn(),
+  })),
+});
+
 describe('ChannelService', () => {
   let service: ChannelService;
+  let dmRepository: MockType<Repository<DirectMessageChannel>>;
 
   beforeEach(async () => {
     const module: TestingModule = await Test.createTestingModule({
       providers: [
         ChannelService,
         { provide: getRepositoryToken(Channel), useFactory: mockChannelRepository },
+        { provide: getRepositoryToken(EventChannel), useFactory: mockEventChannelRepository },
+        { provide: getRepositoryToken(DirectMessageChannel), useFactory: mockDmChannelRepository },
       ],
     }).compile();
 
     service = module.get<ChannelService>(ChannelService);
+    dmRepository = module.get(getRepositoryToken(DirectMessageChannel));
   });
 
   it('should be defined', () => {
     expect(service).toBeDefined();
   });
 
-  it('should get a channel', async () => {
-    const getChannel = await service.getChannel(mockAllChannels[0].id);
-    expect(getChannel).toBe(mockAllChannels[0]);
+  it('should get a generic channel', async () => {
+    const returnedChannel = await service.getChannel(mockAllChannels[0].id);
+    expect(returnedChannel).toBe(mockAllChannels[0]);
+  });
```

Comment:
The way I was thinking about `channel`, is that it would be abstract class and we'd have other channels that implement the interface. Feel like this would be a clean and OO approach. What do you think, do we ever really want to access a channel that doesn't implement the generic one?

Reply:
We will delete this resolver.